.github/workflows/analyzers.yaml

```
@@ -0,0 +1,95 @@
+name: Run analyzers
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  cppcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -y install cppcheck
+
+      - name: Configure
+        run: |
+          cmake -S . -B build                     \
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=ON    \
+            -DBUILD_PYTHON=OFF                    \
+            -DBUILD_TESTING=OFF
+
+      - name: Run cppcheck
+        run: |
+          cppcheck                                          \
+            --enable=style,portability,performance,warning  \
+            --library=posix                                 \
+            --library=cppcheck/segyio.cfg                   \
+            --suppressions-list=cppcheck/suppressions.txt   \
+            --inline-suppr                                  \
+            --project=build/compile_commands.json           \
+            --error-exitcode=1
+
+  scan_build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo DEBIAN_FRONTEND=noninteractive apt-get -y install clang clang-tools libfindbin-libs-perl
+
+      - name: Configure segyio
+        run: |
+          sudo scan-build --status-bugs cmake \
+            -S .                              \
+            -B build                          \
+            -DCMAKE_BUILD_TYPE=Release        \
```

```suggestion
            -DCMAKE_BUILD_TYPE=Debug          \
```
I have no experience with scan-build, but it the [documentation recommends to use a debug build](https://clang.llvm.org/docs/analyzer/user-docs/CommandLineUsage.html#always-analyze-a-project-in-its-debug-configuration).